app/src/main/java/com/arctouch/codechallenge/ui/activity/MovieDetailsActivity.java

```diff
@@ -0,0 +1,139 @@
+    TextView mTxtViewMovieReleaseDate;
+
+    @BindView(R.id.movie_imdb_link)
+    TextView mTxtViewImdbLink;
+
+    @BindView(R.id.progress_bar)
+    SmoothProgressBar mProgressBar;
+
+    MovieDetailsActivityPresenter mMovieDetailsActivityPresenter;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_movie_details);
+
+        ButterKnife.bind(this);
+        mMovieDetailsActivityPresenter = new MovieDetailsActivityPresenterImplementation(this);
+
+        updateProgressBar(true);
+        mMovieDetailsActivityPresenter.start(getIntent());
+    }
+
+    @Override
+    public void showMovieDetails(Movie movie) {
+        //texts
+
+        mTxtViewMovieName.setText(movie.getTitle());
+        mTxtViewMovieGenres.setText(TextUtils.join(", ", movie.getGenres()));
+        String minutesDutarion = movie.getDuration() != null ? movie.getDuration() + getString(R.string.minutes_abbreviation) : ""; //there are some movies with null duration
+        mTxtViewMovieDuration.setText(minutesDutarion);
+        mTxtViewMovieTagline.setText(movie.getTagline());
+        mTxtMovieOverview.setText(movie.getOverview());
+        mTxtViewMovieReleaseDate.setText(movie.getReleaseDate());
+        mTxtViewImdbLink.setText(String.format(Constants.IMDB_MOVIE_URL, movie.getImdbid()));
+
+        //images
+        MovieImageUrlBuilder movieImageUrlBuilder = new MovieImageUrlBuilder();
+
+        String backdropPath = movie.getBackdropPath();
+        if (!TextUtils.isEmpty(backdropPath)) {
+            Glide.with(mImgViewMovieBackDrop)
+                    .load(movieImageUrlBuilder.buildBackdropUrl(backdropPath))
+                    .apply(new RequestOptions().placeholder(R.drawable.ic_image_placeholder)) //TODO ugly ph and it blinks
+                    .into(mImgViewMovieBackDrop);
+        }
+
+        String posterPath = movie.getPosterPath();
+        if (!TextUtils.isEmpty(posterPath)) {
+            Glide.with(mImgViewMoviePoster)
+                    .load(movieImageUrlBuilder.buildPosterUrl(posterPath))
+                    .apply(new RequestOptions().placeholder(R.drawable.ic_image_placeholder)) //TODO ugly ph and it blinks
+                    .into(mImgViewMoviePoster);
+        }
+
+        Objects.requireNonNull(getSupportActionBar()).setDisplayHomeAsUpEnabled(true);
+        getSupportActionBar().setDisplayShowHomeEnabled(true);
+
+        updateProgressBar(false);
+    }
+
+    @Override
+    public void updateProgressBar(boolean show) {
+        if (mProgressBar != null) {
+            mProgressBar.setVisibility(show ? View.VISIBLE : View.INVISIBLE);
+            if (show) {
+                mProgressBar.progressiveStart();
+            } else {
+                mProgressBar.progressiveStop();
+            }
+        }
+    }
+
+    @Override
+    public boolean onSupportNavigateUp() {
+        onBackPressed();
```

Comment:
This is kinda bad. If you had added a `parentActivityName` to this activity in the manifest, it would have been handled automatically.